src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets

```diff
@@ -170,6 +207,10 @@ Copyright (c) .NET Foundation. All rights reserved.
                  ResourceName="ImplicitRuntimeIdentifierResolutionForPublishPropertyFailed"
                  FormatArguments="SelfContained"/>
 
+    <NETSdkError Condition="'$(_PublishSelfContainedIsActive)' == 'true' and '$(PublishSelfContained)' == 'true' and '$(RuntimeIdentifier)' == '' and '$(RuntimeIdentifiers)' == ''"
```

Comment:
Can we merge the changes from https://github.com/dotnet/sdk/pull/28484 first, and follow that pattern in the new condition here (ie no need to check `RuntimeIdentifiers`?

Reply:
I don't think we need this error anymore and can just rely on the `SelfContained` one. Let me know if you still wanna merge that one first.